qnm_filter/utility.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Feb 20 17:55:14 2023
+
+@author: neil
+"""
+
+
+from joblib import Parallel, delayed
+import matplotlib.pyplot as pl
+import numpy as np
+import time
+
+def parallel_compute(function, x_arr, y_arr, kwargs = {}, show_time = True):
```

Comment:
Please don't pass the single function. Instead, pass the whole class instance as `self`; and change `Filter.ipynb` accordingly.

Reply:
For this one, do you want me to make both functions methods of the Network class or do you want me to create a new utility class with these two functions as methods?

Reply:
Neither. The current module is fine. What I was suggesting is to replace `function` with `self` (an instance of the `Network` class. The name can be arbitrary. It's just more conventional to use `self`) and then call `self.likelihood_vs_mass_spin` in the loop.

Reply:
Yeah, it makes sense. It's just that we don't need to type `instance.likelihood_vs_mass_spin` each time we call `parallel_compute` (which might be annoying). We can bring the generalizability back in the future if we really need it.